com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -21,19 +21,122 @@ public enum NetworkAuthority
+            internal const int RotAngleXBit = 4;
+            internal const int RotAngleYBit = 5;
+            internal const int RotAngleZBit = 6;
+            internal const int ScaleXBit = 7;
+            internal const int ScaleYBit = 8;
+            internal const int ScaleZBit = 9;
+            // 10-15: <unused>
+            public ushort Bitset;
+
+            public bool InLocalSpace
+            {
+                get => (Bitset & (1 << InLocalSpaceBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << InLocalSpaceBit);
+            }
+            // Position
+            public bool HasPositionX
+            {
+                get => (Bitset & (1 << PositionXBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << PositionXBit);
+            }
+            public bool HasPositionY
+            {
+                get => (Bitset & (1 << PositionYBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << PositionYBit);
+            }
+            public bool HasPositionZ
+            {
+                get => (Bitset & (1 << PositionZBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << PositionZBit);
+            }
+            // RotAngles
+            public bool HasRotAngleX
+            {
+                get => (Bitset & (1 << RotAngleXBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleXBit);
+            }
+            public bool HasRotAngleY
+            {
+                get => (Bitset & (1 << RotAngleYBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleYBit);
+            }
+            public bool HasRotAngleZ
+            {
+                get => (Bitset & (1 << RotAngleZBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleZBit);
+            }
+            // Scale
+            public bool HasScaleX
+            {
+                get => (Bitset & (1 << ScaleXBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleXBit);
+            }
+            public bool HasScaleY
+            {
+                get => (Bitset & (1 << ScaleYBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleYBit);
+            }
+            public bool HasScaleZ
+            {
+                get => (Bitset & (1 << ScaleZBit)) != 0;
+                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleZBit);
+            }
+
+            public float PositionX, PositionY, PositionZ;
+            public float RotAngleX, RotAngleY, RotAngleZ;
+            public float ScaleX, ScaleY, ScaleZ;
 
             public void NetworkSerialize(NetworkSerializer serializer)
             {
-                serializer.Serialize(ref InLocalSpace);
-                serializer.Serialize(ref Position);
-                serializer.Serialize(ref Rotation);
-                serializer.Serialize(ref Scale);
+                // InLocalSpace + HasXXX Bits
```

Comment:
I wish we already had stateful serialization so that we could just send "same bools as previously" to save on bandwidth. But I guess this will come with snapshot and delta encoding.